app/src/combo.c

```diff
@@ -33,6 +34,8 @@ struct combo_cfg {
     // the virtual key position is a key position outside the range used by the keyboard.
     // it is necessary so hold-taps can uniquely identify a behavior.
     int32_t virtual_key_position;
+    int8_t layers[CONFIG_ZMK_COMBO_MAX_LAYER_FILTERS_PER_COMBO];
+    int32_t layers_len;
```

Comment:
Why use a fixed length array here, if we're also going to need to store a length parameter? We should be able to do something like we do in `app/drivers/kscan/kscan_gpio_direct.c` and make this:

```suggestion
    int32_t layers_len;
    int8_t layers[];
```

If we go this route, we can skip the fixed limit in the Kconfig you added.